File: Leetcode/Python/3542.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        q = []
        ops = 0
        for n in nums:
            while q and q[-1] > n:
                q.pop()
            if n == 0:
                continue
            if not q or q[-1] < n:
                ops += 1
                q.append(n)
        return ops

File: Leetcode/cpp/3542.cpp
class Solution {
public:
  int minOperations(vector<int>& nums) {
    vector<int>queue;
    int ops = 0;
    for (int i = 0; i< nums.size(); ++i) {
      while (!queue.empty() && queue.back() > nums[i]) {
        queue.pop_back();
      }
      if (nums[i] == 0) continue;
      if (queue.empty() || queue.back() < nums[i]) {
        ops++;
        queue.push_back(nums[i]);
      }
    }
    return ops;
  }
};
